src/ORM/Queries/SQLSelect.php

```diff
@@ -564,6 +576,37 @@ public function getUnions(): array
         return $this->union;
     }
 
+    /**
+     * Adds a Common Table Expression (CTE), aka WITH clause.
+     *
+     * Use of this method should usually be within a conditional check against DB::get_conn()->supportsCteQueries().
+     *
+     * @param string $name The name of the WITH clause, which can be referenced in any queries UNIONed to the $query
+     * and in this query directly, as though it were a table name.
+     * @param string[] $cteFields Aliases for any columns selected in $query which can be referenced in any queries
+     * UNIONed to the $query and in this query directly, as though they were columns in a real table.
+     */
+    public function addWith(string $name, self $query, array $cteFields = [], bool $recursive = false): static
```

Comment:
```suggestion
    public function addWith(string $name, SQLSelect $query, array $cteFields = [], bool $recursive = false): static
```
Just make it a little easier to read

Reply:
Done - you allowed `self` as the typehint for union in a recent PR so I've changed that as well to be consistent.